go/cmd/vtcombo/main.go

```diff
@@ -275,3 +274,36 @@ func main() {
 	})
 	servenv.RunDefault()
 }
+
+// vtcomboMysqld is a wrapper on top of mysqlctl.Mysqld.
+// We need this wrapper because vtcombo runs with a single MySQL instance
+// which all the tablets connect to. (replica, primary, all). This means that we shouldn't
+// be trying to run any replication related commands on it, otherwise they fail.
```

Comment:
While looking at the test failures, I found a couple using vtcombo and discovered that it was not starting at all.
On further investigation, I found that vtcombo runs with 1 MySQL instance so running replication commands makes it fail.
Vtcombo started failing in initialization because we are now trying to set replication source on startup in vttablets.

The problem has been resolved by adding a wrapper on mysqld to ignore replication-related rpcs.

This change in my opinion, is large enough to require a new review.